geopandas/io/file.py

```diff
@@ -400,7 +400,15 @@ def infer_schema(df):
     from collections import OrderedDict
 
     # TODO: test pandas string type and boolean type once released
-    types = {"Int64": "int", "string": "str", "boolean": "bool"}
+    # "Int64":"int" is used by pandas for conversions
+    # "int32":"int32" used to ensure that int32 is used instead of regular int
+    types = {
+        "Int64": "int",
+        "int64": "int32",
```

Comment:
Is there a reason this is also needed? 

So my understanding currently is that when we use "int" here, fiona will automatically use this as "int64" (if your GDAL is recent enough to support int64). And only if explicitly mentioning "int32", it will use the int32 gdal type. 
But so for the drivers that support int64 types, we should continue to use that by default. But when your data has an "int32" column, we should respect that and let fiona write this as int32 to the file. This way, you have a workaround possible if you write to a file that doesn't support int64 by manually casting to int32 before writing to file.

Reply:
Additional thought: if you have int8 or int16 columns originally, we should maybe also write them as int32 instead of int64.